dev_suites/dev_options_suite/options_suite.rb

```diff
@@ -0,0 +1,99 @@
+module OptionsSuite
+  class V1Group < Inferno::TestGroup
+    title 'V1 Group'
+    id :v1_group
+
+    test do
+      title 'V1 Test'
+      id :v1_test
+
+      run do
+        assert suite_options[:ig_version] == '1'
+      end
+    end
+  end
+
+  class V2Group < Inferno::TestGroup
+    title 'V2 Group'
+    id :v2_group
+
+    test do
+      title 'V2 Test 1'
+      id :v2_test1
+
+      run do
+        assert suite_options[:ig_version] == '2'
+      end
+    end
+
+    test do
+      title 'V2 Test 2'
+      id :v2_test2
+
+      run do
+        assert suite_options[:ig_version] == '2'
+      end
+    end
+  end
+
+  class AllVersionsGroup < Inferno::TestGroup
+    title 'All Versions Group'
+    id :all_versions_group
+
+    test do
+      title 'All Versions Test 1'
+      id :all_versions_test1
+
+      run { pass }
+    end
+
+    test do
+      title 'All Versions Test 2'
+      id :all_versions_test2
+
+      run { pass }
+    end
+
+    test do
+      title 'All Versions Test 3'
+      id :all_versions_test3
+
+      run { pass }
+    end
+  end
+
+  class Suite < Inferno::TestSuite
+    title 'Options Suite'
+    id :options
+
+    validator required_suite_options: { ig_version: '1' } do
+      url 'v1_validator'
+    end
+
+    validator required_suite_options: { ig_version: '2' } do
+      url 'v2_validator'
+    end
+
+    suite_option :ig_version,
+                 title: 'IG Version',
+                 description: 'Which IG Version should be used',
+                 list_options: [
+                   {
+                     label: 'v1',
+                     value: '1'
+                   },
+                   {
+                     label: 'v2',
+                     value: '2'
+                   }
+                 ]
+
+    group from: :all_versions_group
+
+    group from: :v1_group,
+          required_suite_options: { ig_version: '1' }
+
+    group from: :v2_group,
+          required_suite_options: { ig_version: '2' }
```

Comment:
Is it expected that I see both V1 & V2 Test Groups in the UI? I was expecting to see none because the IG version isn't handled/set by the UI/API.

![Screen Shot 2022-06-10 at 1 08 52 PM](https://user-images.githubusercontent.com/41651655/173117042-cb736815-45f9-4806-921f-6d910a7baf86.png)

Reply:
Oh I'm guessing this is because of the children method in `runnable`.

```ruby
def children(selected_suite_options)
        return all_children if selected_suite_options.blank?

        all_children.select do |child|
          requirements = child.suite_option_requirements || {}

          # requirements are a subset of selected options or equal to selected options
          selected_suite_options >= requirements
        end
      end
```

It seems odd that these tests would default to showing if their required suite_option isn't present. I think it would make more sense to still filter children by their requirements and if the parent has no requirements, then only children with no required suite options are included.

Reply:
Yes, the JSON API and front end don't know anything about suite options, so they don't do any filtering.

Reply:
It defaults to showing everything because if you just ask for the suite, you're going to get the entire suite. You'll only get the filtered version in the context of a test session, and the front end has no way to communicate that until the JSON API is in place.

Reply:
So, the way the JSON API will work is basically:

- `GET api/test_suites/:id` - get the whole unfiltered test suite
- `POST api/test_sessions or GET api/test_sessions/:id` - the response body contains the filtered test suite based on the selected options